### PR TITLE
 Add Logging to Callback Server AND change max memory

### PR DESCRIPTION
**Pull request by bio-boris**

### CODE REVIEW

**Comment by scanon** [comment]
I think this will create too many files, but I like the general idea.

How about create a single file in the constructor and then add a logging function that will append to that single file.

**Comment by MrCreosote** [comment]
Is this a temp thing for debugging?

**Comment by bio-boris** [comment]
@scanon now it prints to one log file

**Comment by scanon** [comment]
I think this is good.

**Comment by MrCreosote** [comment]
Note the server is multithreaded so there may be issues with logs interleaving depending on how the code calls subjobs.

File: src/us/kbase/common/executionengine/CallbackServer.java [hunk 1]
```diff
@@ -1,10 +1,6 @@
 package us.kbase.common.executionengine;
 
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.io.OutputStream;
-import java.io.PrintWriter;
-import java.io.StringWriter;
+import java.io.*;
```
Comment (MrCreosote) [inline src/us/kbase/common/executionengine/CallbackServer.java]: This makes it difficult to figure out where a name comes from - explicit imports are more readable imo

File: src/us/kbase/common/executionengine/CallbackServer.java [hunk 2]
```diff
@@ -241,9 +241,12 @@ protected void processRpcCall(RpcCallData rpcCallData, String token, JsonServerS
         final ModuleMethod modmeth = new ModuleMethod(
                 rpcCallData.getMethod());
         final Map<String, Object> jsonRpcResponse;
-        
+
+
         if (modmeth.isCheck()) {
             jsonRpcResponse = runCheck(rpcCallData);
+            String logString = String.format("BEGIN_IS_CHECK_LOG|RPCDATA:%s|RPCRESPONSE:%s|TIMESTAMP:%s|END_IS_CHECK_LOG", rpcCallData.toString(), jsonRpcResponse.toString(),System.currentTimeMillis());
```
Comment (MrCreosote) [inline src/us/kbase/common/executionengine/CallbackServer.java]: This is going to be really hard to read - I'd probably pretty print the json on each line.

Comment (MrCreosote) [reply src/us/kbase/common/executionengine/CallbackServer.java]: wait that won't work due to interleaving

Comment (MrCreosote) [reply src/us/kbase/common/executionengine/CallbackServer.java]: I'd at least put the different json strings on different lines, using the job id to link up the lines.

**Comment by MrCreosote** [comment]
Looks fine otherwise.

File: src/us/kbase/common/executionengine/CallbackServer.java (hunk 2 above)
Comment (bio-boris) [reply src/us/kbase/common/executionengine/CallbackServer.java]: Yea it looks pretty bad just checked it, will try a nicer format

File: src/us/kbase/common/executionengine/CallbackServer.java (hunk 1 above)
Comment (bio-boris) [reply src/us/kbase/common/executionengine/CallbackServer.java]: This is the result of the optimize imports button. Usually I just cmd click a variable in intellij and it tells me which package it came from.

Comment (MrCreosote) [reply src/us/kbase/common/executionengine/CallbackServer.java]: Which you don't have when you're doing github reviews or working in a text editor

Comment (bio-boris) [reply src/us/kbase/common/executionengine/CallbackServer.java]: True, good point.  I'm used to cloning repos, loading them up in my IDE and running the tests locally, but it is nice when Travis runs the tests for you.

**Pull request merged**